chia/consensus/default_constants.py

```diff
@@ -36,10 +36,8 @@
         "3d8765d3a597ec1d99663f6c9816d915b9f68613ac94009884c4addaefcce6af"
     ),
     "MAX_VDF_WITNESS_SIZE": 64,
-    # Target tx count per sec
-    "TX_PER_SEC": 20,
-    # Size of mempool = 10x the size of block
-    "MEMPOOL_BLOCK_BUFFER": 10,
+    # Size of mempool = 5x the size of block
+    "MEMPOOL_BLOCK_BUFFER": 5,
```

Comment:
@mariano54 Let's increase this to 150x size of blocks, this will be just over 2H worth of transactions.  (I discussed this with Bram)

Reply:
My fear is that this 150x would make DOS attacks much easier because:
1. You can send transactions for 2 hours without really paying any fees and spam the whole network, as opposed to just a few minutes
2. If you have shared coins (like a uniswap coin that many people use), then there is an attack, where the attacker constantly fills the uniswap contract to like half of the block size, but with a low fee, so it won't get into the block (but not low enough that it get's kicked out of mempool). If the mempool is small, this is not a big deal, because that TX will either get included or get kicked out. With a large mempool, that tx might be stuck there for a long time, which means that attacker can easily censor the uniswap contract, or force people to pay for his fees.

Reply:
Also is there a benefit from such a high value? It seems safer to just go with the conservative number now and increase it later

Reply:
I increased to 150x for the time being. But let me know what you think about the above attacks.

Reply:
A greedy ("rational") farmer will maximize revenue for itself, regardless of what we might think healthy for the network, and even what code we choose to ship (since they could patch their own local node to "do better" if they want). So we may as well try to maximize revenue for the farmer, even if that means replacing low-fee-per-cost uniswap spends with later-but-higher-fee-per-cost uniswap spends.

Reply:
Yeah we want to always replace tx with the highest fee-per-cost. However the attack that I'm taking about is when the attacker purposefully doesn't put a high enough fee to get included (so no revenue for farmer).

Still, I think it's in the farmers best interest to have a large mempool, in case it gets cleared quickly. So that's an argument for a big mempool.

Reply:
On a more general note, I think it's pretty rational for a full mempool to eject its lowest cost-per-fee coin spend when a higher one comes along. It's pretty much impossible to prevent frontrunners from bribing market-makers. That's why people want to be market-makers!

Reply:
> Yeah we want to always replace tx with the highest fee-per-cost. However the attack that I'm taking about is when the attacker purposefully doesn't put a high enough fee to get included (so no revenue for farmer).
> 
> Still, I think it's in the farmers best interest to have a large mempool, in case it gets cleared quickly. So that's an argument for a big mempool.

Not sure I get it... seems like the attacker's uniswap tx would get kicked out the mempool by other higher-paying txs with a big or small mempool. What am I missing?

Reply:
We have a mechanism where in order to replace a spend, you have to spend the same coins, and optionally more coins, you cannot spend less coins, in order to prevent DOS attacks.

I'm thinking of a case where there is a uniswap coin U and user coins A1....A500.
In order to do  trade, you spend U and A1 in a bundle.

Lets say the attacker spends a bundle with U and A5 ... A500. (with a low fee per cost).

The user can not remove the A5....A500 spends now. The user can add their own spend of A1, but the fee per cost would not increase much, because it's a huge transaction. In orderf to get the TX included, the user would have to bump the fee for the whole (large) transaction, which would probably cost 500x more than a normal fee. If the mempool is smaller, then it should be easier to bump it into inclusion, do you see what i'm saying?

Reply:
Although on second thought, 2 hours worth of mempool is not too much, considering bitcoin has over a day of transaction capacity in the mempool.

Reply:
> We have a mechanism where in order to replace a spend, you have to spend the same coins, and optionally more coins, you cannot spend less coins, in order to prevent DOS attacks.
> 
> I'm thinking of a case where there is a uniswap coin U and user coins A1....A500.
> In order to do trade, you spend U and A1 in a bundle.
> 
> Lets say the attacker spends a bundle with U and A5 ... A500. (with a low fee per cost).
> 
> The user can not remove the A5....A500 spends now. The user can add their own spend of A1, but the fee per cost would not increase much, because it's a huge transaction. In orderf to get the TX included, the user would have to bump the fee for the whole (large) transaction, which would probably cost 500x more than a normal fee. If the mempool is smaller, then it should be easier to bump it into inclusion, do you see what i'm saying?

If there's a spend of U that also spends, say, B1 but has a higher fee per cost than the spend with the As, it seems to me that a rational farmer would include the B spend instead of the A spend.